src/imitation/rewards/serialize.py

```diff
@@ -0,0 +1,99 @@
+"""Load serialized policies of different types."""
+
+from typing import Callable
+
+import numpy as np
+from stable_baselines.common.vec_env import VecEnv
+
+from imitation.rewards import discrim_net, reward_net
+from imitation.util import registry, util
+
+RewardFn = Callable[[np.ndarray, np.ndarray, np.ndarray], np.ndarray]
+RewardLoaderFn = Callable[[str, VecEnv], RewardFn]
+RewardNetLoaderFn = Callable[[str, VecEnv], reward_net.RewardNet]
+
+reward_net_registry: registry.Registry[RewardNetLoaderFn] = registry.Registry()
+reward_registry: registry.Registry[RewardLoaderFn] = registry.Registry()
+
+
+def _add_reward_net_loaders(classes):
+  for name, cls in classes.items():
+    loader = registry.build_loader_fn_require_path(cls.load)
+    reward_net_registry.register(key=name, value=loader)
+
+
+REWARD_NETS = {
+    "BasicRewardNet": reward_net.BasicRewardNet,
+    "BasicShapedRewardNet": reward_net.BasicShapedRewardNet,
+}
+_add_reward_net_loaders(REWARD_NETS)
+
+
+def _load_discrim_net(cls):
+  def f(path: str, venv: VecEnv):
+    discrim_net = cls.load(path)
+    # TODO(gleave): expose train reward as well? (hard due to action probs?)
+    return discrim_net.reward_test
+  return f
+
+
+def _add_discrim_net_loaders(classes):
+  for name, cls in classes.items():
+    reward_registry.register(key=name, value=_load_discrim_net(cls))
+
+
+_add_discrim_net_loaders({
+    "DiscrimNetAIRL": discrim_net.DiscrimNetAIRL,
+    "DiscrimNetGAIL": discrim_net.DiscrimNetGAIL,
+})
+
+
+def load_reward_net_as_fn(path: str, env: VecEnv) -> RewardFn:
+  reward_type, shaped, reward_path = path.split(':')
```

Comment:
I don't want to add a `reward_shaped` parameter to Sacred since many reward models won't have a shaped/unshaped variant. This is just specific to `RewardNet`, and really just to AIRL.

I've changed this to make the `reward_type` (top-level, in `load_reward`) be either `RewardNet_unshaped` or `RewardNet_shaped`, and then pass a `shaped` term to `load_reward_net_as_fn`.

So the end user would specify e.g. `with reward_type=RewardNet_unshaped reward_path=BasicShapedRewardNet:~/output/train/path/to/reward_net`

Reply:
That seems ok 👌